semgrep-core/src/analyzing/AST_to_IL.ml

```diff
@@ -352,16 +372,37 @@ and assign env lhs _tok rhs_exp eorig =
 (* less: we could pass in an optional lval that we know the caller want
  * to assign into, which would avoid creating useless fresh_var intermediates.
  *)
-and expr_aux env eorig =
+and expr_aux env ?(void = false) eorig =
   match eorig.G.e with
-  | G.Call ({ e = G.IdSpecial (G.Op op, tok); _ }, args) ->
+  | G.Call ({ e = G.IdSpecial (G.Op op, tok); _ }, args) -> (
       let args = arguments env args in
-      mk_e (Operator ((op, tok), args)) eorig
+      if not void then mk_e (Operator ((op, tok), args)) eorig
+      else
```

Comment:
Are there any operators of this kind which return something and also mutate state? (This seems like a useful proxy, I'm more just curious)

Reply:
Probably yes. I hope stdlibs do not do that, but users can totally define such kind of operators.